lib/cppcheck.cpp

```diff
@@ -674,7 +688,13 @@ unsigned int CppCheck::checkFile(const FileWithDetails& file, const std::string
                     tokenizer.list.createTokens(in, file.spath());
                 }
                 mUnusedFunctionsCheck->parseTokens(tokenizer, mSettings);
-                // TODO: set analyzer information
+
+                // TODO: how to get the proper tokenlist to generate the proper hash?
+                const std::size_t hash = time(nullptr); // calculateHash(tokenizer.list, mSettings);
```

Comment:
This requires a `simplecpp::TokenList` which we do not create in this context.

This is a hack so the hash is different every time.

Reply:
The Preprocessor::calculateHash is pretty simple it should be possible to write a similar function that uses `tokenizer.list`.. as shown in the comment?

Reply:
I can have a look.

Reply:
I now generate a hash from the proper inputs.